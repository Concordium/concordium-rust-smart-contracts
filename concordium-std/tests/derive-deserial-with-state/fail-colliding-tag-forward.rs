//! Ensure `derive(DeserialWithState)` fails when 'forward' and 'tag' attributes
//! are colliding.
use concordium_std::*;

#[derive(DeserialWithState)]
#[concordium(state_parameter = "S", repr(u8))]
enum Count<S> {
    One {
        field: StateSet<u32, S>,
    },
    #[concordium(forward = [5, 6])]
    Two(Inner),

    #[concordium(tag = 5)]
    Three(Inner),
}

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Inner {
    #[concordium(tag = 5)]
    Alpha {
        balance: u32,
    },
    #[concordium(tag = 6)]
    Beta(u16),
}

fn main() {}
