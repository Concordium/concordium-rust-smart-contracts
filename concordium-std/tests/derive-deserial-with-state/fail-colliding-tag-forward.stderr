error: The tag of 'Three' collides with the tag of 'Two'.
  --> tests/derive-deserial-with-state/fail-colliding-tag-forward.rs:14:18
   |
14 |     #[concordium(tag = 5)]
   |                  ^^^

error: The tag of 'Two' collides with the tag of 'Three'.
  --> tests/derive-deserial-with-state/fail-colliding-tag-forward.rs:11:29
   |
11 |     #[concordium(forward = [5, 6])]
   |                             ^
